jax/interpreters/pxla.py

```diff
@@ -331,32 +384,42 @@ class ShardedDeviceArray(xla.DeviceArray):
   behavior of an ndarray so that it can be treated by user code as an ndarray;
   that is, it is only an optimization to reduce transfers.
 
-  The number of device buffers underlying a ShardedDeviceArray instance is equal
-  to the number of replicas of the computation that produced it. Each buffer
-  represents a shard of the original array, meaning a slice along its leading
-  axis. These component buffers reside on distinct devices, but need not
-  represent distinct logical shards. The correspondence can be computed with
-  the assign_shards_to_replicas function.
+  Attributes:
+    aval: A ShapedArray indicating the shape and dtype of this array.
+    sharding_spec: describes how this array is sharded across `device_buffers`.
+    device_buffers: the buffers containing the data for this array. Each buffer
+      is the same shape and on a different device. Buffers are in row-major
+      order, with replication treated as an extra innermost dimension. For
+      example, if the unsharded data is [1, 2], the replicated data should be
+      laid out here as [1, 1, 2, 2].
```

Comment:
I do not understand where the new shape is coming from. Is this the shape on the device? What sharding_spec is this assuming? As I commented above, I thought that replication is considered as the leading axis, and is squeezed, so it does not show in the shape of the buffers. 

Nit: Technically, shapes are tuples and you write them as lists in this comment.

Reply:
This example apparently wasn't clear at all. The lists are supposed to be the actual array values, not shapes. I just removed this last sentence since it only added confusion.